doc/source/serve/tutorials/batch.rst

```diff
@@ -26,51 +26,52 @@ Let's import Ray Serve and some other helpers.
+The function must handle a list of objects and will be called with a single object.
+This function must also be ``async def`` so that you can handle multiple queries concurrently:
 
 .. code-block:: python
 
-    def single_request(
-        request: Union[starlette.requests.Request, ServeRequest],
-    ):
+    @serve.batch
+    async def my_batch_handler(self, requests: List):
         pass
 
-For batched backends, the input types are converted to list of their original
-types:
+This batch handler can then be called from another ``async def`` method in your backend.
+These calls will be batched and executed together, but return an individual result as if
+they were a normal function call:
 
 .. code-block:: python
 
-    @serve.accept_batch
-    def batched_request(
-        request: List[Union[starlette.requests.Request, ServeRequest]],
-    ):
-        pass
+    class MyBackend:
+        @serve.batch
+        async def my_batch_handler(self, requests: List):
+            results = []
+            for request in requests:
+                results.append(request.json())
+            return results
 
-Let's define the backend function. We will take in a list of requests, extract
-the input value, convert them into an array, and use NumPy to add 1 to each element.
+        async def __call__(self, request):
+            await self.my_batch_handler(request)
+
+Let's define a backend that takes in a list of requests, extracts the input value,
+converts them into an array, and uses NumPy to add 1 to each element.
 
 .. literalinclude:: ../../../../python/ray/serve/examples/doc/tutorial_batch.py
-    :start-after: __doc_define_servable_v0_begin__
-    :end-before: __doc_define_servable_v0_end__
+    :start-after: __doc_define_servable_begin__
+    :end-before: __doc_define_servable_end__
 
-Let's deploy it. Note that in the ``config`` section of ``create_backend``, we
-are specifying the maximum batch size via ``config={"max_batch_size": 4}``. This
-configuration option limits the maximum possible batch size sent to the backend.
+Let's deploy it. Note that in the ``@serve.batch`` decorator, we are specifying
+specifying the maximum batch size via ``max_batch_size=4``. This option limits
+the maximum possible batch size that will be executed at once.
 
 .. note::
-    Ray Serve performs *opportunistic batching*. When a replica is free to evaluate
-    the next batch, Ray Serve will look at the pending queries and try to take
-    :mod:`max_batch_size <ray.serve.BackendConfig>` queries to form a batch.
-    If the batch has less than ``max_batch_size`` queries, the backend will
-    wait for :mod:`batch_wait_timeout <ray.serve.BackendConfig>`
-    seconds to wait for a full batch to arrive. The default wait is ``0s`` to
-    minimize query latency. You can increase the timeout to improve throughput
-    and increase utilization at the cost of some additional latency.
+    By default, Ray Serve performs *opportunistic batching*. This means that as
```

Comment:
This note was the most useful piece of the documentation for me when I was first reading it (though maybe it's very standard and not that useful to others).  Is there a way to make it more prominent/move it higher up without disrupting the flow?

Reply:
I think it might be fine to just put it all the way at the top after the top-level description of how to do batching. Thoughts?

Reply:
I moved it just after the initial code sample, I think that's an improvement. Any earlier is probably confusing. LMK what you think.